vividus-extension-visual-testing/src/main/java/org/vividus/visual/AbstractVisualCheckFactory.java

```diff
@@ -21,36 +21,20 @@
 
 import javax.inject.Inject;
 
-import org.vividus.ui.screenshot.ScreenshotConfiguration;
-import org.vividus.ui.screenshot.ScreenshotParametersFactory;
-import org.vividus.visual.model.AbstractVisualCheck;
 import org.vividus.visual.screenshot.IScreenshotIndexer;
 
-public abstract class AbstractVisualCheckFactory<T extends AbstractVisualCheck>
+public abstract class AbstractVisualCheckFactory
 {
     private Map<String, IScreenshotIndexer> indexers;
     private Optional<String> screenshotIndexer;
 
-    private final ScreenshotParametersFactory<ScreenshotConfiguration> screenshotParametersFactory;
-
-    protected AbstractVisualCheckFactory(
-        ScreenshotParametersFactory<ScreenshotConfiguration> screenshotParametersFactory)
-    {
-        this.screenshotParametersFactory = screenshotParametersFactory;
-    }
-
     protected String createIndexedBaseline(String baselineName)
     {
         return screenshotIndexer.map(indexers::get)
                 .map(indexer -> indexer.index(baselineName))
                 .orElse(baselineName);
     }
```

Comment:
This logic should be extracted into a new class (e.g. BaselineIndexer), then this class should be reused and `AbstractVisualCheckFactory` should be removed